spec/unit/changelog_source_spec.rb

```diff
@@ -0,0 +1,182 @@
+# Copyright (c) 2013-2015 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 3 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE about this file by physical or electronic mail,
+# you may find current contact information at www.suse.com
+
+require_relative "../../lib/changelog_source"
+
+describe ChangelogSource do
+  let(:file) { "" }
+
+  subject { ChangelogSource.new(file) }
+
+  describe "#append_item_to_element" do
+    let(:fresh_element) {
+      {
+        version: "1.1.1",
+        date: "Fri Mar 13 13:03:47 CET 2015",
+        email: "johndoe@example.com",
+        items: []
+      }
+    }
+    let(:item) { "Fix issue with changes of managed files on RHEL (gh#SUSE/machinery#636)" }
+
+    before do
+      ChangelogSource.append_item_to_element(fresh_element, item)
+    end
+
+    it "inserts the given element into the items array" do
+      expect(fresh_element[:items]).not_to be_empty
+    end
+
+    # it "inserts the given element into the items array" do
+    #   expect(subject.elements).to include()
+    # end
+  end
+
+  describe "#parse_item" do
+    context "when the item is written in a single line" do
+      let(:dirty_item) {
+        " Fix issue with changes of managed files on RHEL (gh#SUSE/machinery#636)"
+      }
+
+      it "returns an item" do
+        expect(ChangelogSource.parse_item(dirty_item))
+          .to eq("Fix issue with changes of managed files on RHEL (gh#SUSE/machinery#636)")
+      end
+    end
+
+    context "when the item is written in multiple lines" do
+      let(:dirty_item) {
+        <<-EOF
+  Add experimental `--exclude` option to exclude elements from inspection.
+  Run `machinery config experimental-features on` to enable the option
+EOF
+      }
+
+      it "returns an item" do
+        expect(ChangelogSource.parse_item(dirty_item))
+          .to eq("Add experimental `--exclude` option to exclude elements from inspection.\n" \
+                 "  Run `machinery config experimental-features on` to enable the option")
+      end
+    end
+  end
+
+  describe "#parse_element" do
+    context "with a valid version line" do
+      let(:valid_version) { "## Version 1.6.0 - Wed Apr 01 15:43:17 CEST 2015 - johndoe@suse.de" }
+      it "returns an element" do
+        expect(ChangelogSource.parse_element(valid_version))
+          .to eq(version:  "1.6.0",
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Unnecessary spacing detected.